### PR TITLE
Fix #503 : Question player activity test

### LINKED ISSUE #503: Add thorough tests for the question player
Similar to #388 we need thorough tests for the question player once these can be written in a reliable way for both Espresso & Robolectric.

Test reference: https://github.com/oppia/oppia-android/blob/develop/app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt

### PR DESCRIPTION
**Pull request by chrk2205**
Fixes #503 

Draft PR for Question Player Activity Test
## Checklist

- [ ] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by chrk2205** [comment]
@anandwana001 
added some basic test cases

**Comment by chrk2205** [comment]
@anandwana001 
`testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog_clickCancel_dismissesDialog`
for this testcase how do I check if the activity is getting closed or not 
thanks

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 1]
```diff
@@ -147,6 +153,64 @@ class QuestionPlayerActivityTest {
...
+            withText(R.string.question_player_title)
+          )
+        )
+      )
+
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_onConfigurationChange_toolbarTitle_isDisplayedSuccessfully() {
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: use `configChange` in place of `onConfigurationChange`

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 2]
```diff
@@ -147,6 +153,64 @@ class QuestionPlayerActivityTest {
     testCoroutineDispatchers.unregisterIdlingResource()
   }
 
+  @Test
+  fun testQuestionPlayer_toolbarTitle_isDisplayedSuccessfully() {
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: let's go with only `isDisplayed()`

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 3]
```diff
@@ -147,6 +153,64 @@ class QuestionPlayerActivityTest {
...
+          hasDescendant(
+            withText(R.string.question_player_title)
+          )
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog() {
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: stopQuestionPlayerDialogIsDisplayed

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @chrk2205 

As per my [earlier comment](https://github.com/oppia/oppia-android/issues/503#issuecomment-771430549), I am thinking to first cover all the edge cases what tests we need before writing test cases.

```
// TODO(#503): Add more test-cases
  //  1. verifying the toolbar title
  //  2. Dialog appears on back press or clicking on cross mark in toolbar
```

Same as what you are doing here, current added tests look good to me. 


`testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog_clickCancel_dismissesDialog `, if you want to test if dialog is displayed or not, you can use `not(IsDisplayed)` or `doesNotExist` function to check if view is visible or not, or view is in the view hierarchy or not.

**Comment by anandwana001** [comment]
@chrk2205 please complete all the installation process - https://github.com/oppia/oppia-android/wiki#installation

**Comment by chrk2205** [comment]
> @chrk2205 please complete all the installation process - https://github.com/oppia/oppia-android/wiki#installation

@anandwana001 
`oppia-android-tools` is generated but I am not sure how to use it??

**Comment by chrk2205** [comment]
for some edge cases
shall i include the Submit button cannot be clicked if there is no input?

questionplayeractivity

**Comment by anandwana001** [comment]
> > @chrk2205 please complete all the installation process - https://github.com/oppia/oppia-android/wiki#installation
> 
> @anandwana001
> `oppia-android-tools` is generated but I am not sure how to use it??

You don't need to use it, all you need is to run the setup script.

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 4]
```diff
@@ -147,6 +153,64 @@ class QuestionPlayerActivityTest {
...
+  fun testQuestionPlayer_toolbarTitle_isDisplayedSuccessfully() {
+    launchForSkillList(SKILL_ID_LIST).use {
+      onView(withId(R.id.question_player_toolbar)).check(
+        matches(
+          hasDescendant(
+            withText(R.string.question_player_title)
+          )
+        )
+      )
+
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt:221]: Remove this extra blank line

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 5]
```diff
@@ -147,6 +153,64 @@ class QuestionPlayerActivityTest {
...
+      rotateToLandscape()
+      onView(withText(R.string.stop_exploration_dialog_title)).inRoot(isDialog())
+        .check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testQuestionPlayer_onToolbarClosePressed_showsStopQuestionPlayerDialog() {
+    launchForSkillList(SKILL_ID_LIST).use {
+      onView(ViewMatchers.withContentDescription(R.string.nav_app_bar_navigate_up_description)).perform(
```
Comment (rt4914) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: Update:
`onView(withContentDescription(R.string.nav_app_bar_navigate_up_description)).perform(click())`

**Review by rt4914 — CHANGES_REQUESTED** [review]
@chrk2205 PTAL

**Comment by chrk2205** [comment]
@anandwana001  
the activity is launching for specific test skills set.
how do I know what question be displayed ??
as this will determine the type of answers like text or multiple choice.
can you help me find it?
thanks

**Comment by chrk2205** [comment]
how to test if the present activity has closed?
for `testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog_clickOk_activityCloses`
tried using this - `intended(hasComponent(TopicActivity::class.java.name))`

and also if the dialog is disappeared `testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog_clickCancel_dismissesDialog`
tried this - `onView(withText(R.string.stop_exploration_dialog_title)).inRoot(isDialog())
        .check(doesNotExist()) `
thankyou

**Comment by anandwana001** [comment]
@chrk2205 as per my earlier suggestion I suggest not to go for writing test cases as the first step but to collect what all things we need to test in a document or some comments. 

Also, before starting we need to figure out if we need to have a test activity like `StateFragmentTestActivity`, which is used for `StateFragmentTest` because we cannot test the explorations directly so we create a copy and test. 
We need to figure out if we need the same for our `QuestionPlayerActivity`.

Just like `StateFragmentTest`, we are going to use a pre defined set of questions so we can walk them through while testing each phase like using `TEST_SKILL_ID_0` or `SKILL_ID_LIST`. 


<pre>
  /**
   * <b>Tests for QuestionPlayerActivityTest</b>
   * testQuestionPlayer_toolbarIsVisible
   * testQuestionPlayer_toolbarTitleIsCorrect - match the string @string/question_player_title
   * testQuestionPlayer_backPress_stopExplorationDialogIsDisplayed
   * testQuestionPlayer_backPress_clickCancel_dialogIsDismissed
   * testQuestionPlayer_backPress_clickLeave_questionPlayerIsClosed
   * testQuestionPlayer_clickToolbarNav_stopExplorationDialogIsDisplayed
   * testQuestionPlayer_clickToolbarNav_clickCancel_dialogIsDismissed
   * testQuestionPlayer_clickToolbarNav_clickLeave_questionPlayerIsClosed
   * testQuestionPlayer_submitButtonIsDisplayed
   * testQuestionPlayer_configChange_submitButtonIsDisplayed
   * testQuestionPlayer_enterAnswer_submitButtonIsClickable
   * testQuestionPlayer_enterAnswer_configChange_submitButtonIsClickable - this needs to solved, not working currently
   * testQuestionPlayer_submitIncorrectAnswer_submitButtonIsDisabled
   * testQuestionPlayer_submitIncorrectAnswer_configChange_submitButtonIsDisabled
   * .
   * .
   * .
   * .
   * 
   */
</pre>

So, I suggest here to fill this first by looking at what all things need to be tested. You can take reference from `StateFragmentTest`.

**Comment by chrk2205** [comment]
Ok 
I will add all the test names required for this activity.

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt (hunk 3 above)
Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: will do in my next commit

**Comment by chrk2205** [comment]
@anandwana001 
My code is not getting pushed 
By any chance is this because of the post installation process ?
Thanks

**Comment by anandwana001** [comment]
This is possibly because of ktlint errors. 
If you are using command line to push, you will be able to see what are the errors. If you are using Android Studio to push, there will be a small pop up on right bottom side where you can see the errors. 
Also, you can take reference from here https://github.com/oppia/oppia-android/wiki/Ktlint-Guide

**Comment by anandwana001** [comment]
These are the ktlint error you have 

/home/runner/work/oppia-android/oppia-android/app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt:166:1: Unexpected blank line(s) before "}"
/home/runner/work/oppia-android/oppia-android/app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt:206:1: Exceeded max line length (100) (cannot be auto-corrected)

**Comment by chrk2205** [comment]
> This is possibly because of ktlint errors.
> If you are using command line to push, you will be able to see what are the errors. If you are using Android Studio to push, there will be a small pop up on right bottom side where you can see the errors.
> Also, you can take reference from here https://github.com/oppia/oppia-android/wiki/Ktlint-Guide

it's not showing any Ktlint errors. maybe I am not seeing it,
The commands mentioned in the wiki are not working because ktlint is not installed 
where can i find the installation guide for windows as it is not mentioned in 
@anandwana001

**Comment by anandwana001** [comment]
All commands are mentioned in the wiki guide -https://github.com/oppia/oppia-android/wiki/Ktlint-Guide
Also, as per learning purpose also we suggest using the command line from where if you try to push and if there is any ktlint error, the code will not get pushed and show you what to fix. 

Adding @FareesHussain for any help around the command line.

**Comment by FareesHussain** [comment]
@chrk2205 Just for conformation have you done the following 
- Did you run the setup.sh as mentioned in the ktlint-guide (downloads ktlint and add it to `oppia-android-tools`) Reference [4th step in installation](https://github.com/oppia/oppia-android/wiki#installation)
    - Incase You need to run ktlint you can add the path of `oppia-android-tools` as a environment variable



And how are you trying to push the code using the command line or Android Studio (GUI).
It would be better if you can add the screenshot of the error that appeared when you try to push the code for both cmd line and the android studio

**Comment by anandwana001** [comment]
Running script will do all the other work like downloading ktlint and all, no extra work needs to be done by contributors.

**Comment by chrk2205** [comment]
@FareesHussain 
`oppia-android-tools` is created after that am I supposed to do anything??
about the error i am not sure what happened but it worked properly after some changes

**Comment by FareesHussain** [comment]
> @FareesHussain
> `oppia-android-tools` is created after that am I supposed to do anything??
> about the error i am not sure what happened but it worked properly after some changes

It is a bit hard to fix this without knowing how the error is reproduced and what the error was
The following might be the causing it
 - ktlint fixes were not commited before pushing the code
 - Pushing code related to workflow (this might not be the case here)

**Comment by chrk2205** [comment]
@FareesHussain 
about the ktlint error where can I find the Errors?
i mean the changes I need to do so it can pass the Ktlint

**Comment by FareesHussain** [comment]
You can find the errors when you push using the command line.


If you want to setup ktlint, you have to create an environment variable for the `oppia-android-tools` directory.
Then run the command (using git bash) mentioned [here](https://github.com/oppia/oppia-android/wiki/Ktlint-Guide) check the ktlitn manually

**Comment by chrk2205** [comment]
@FareesHussain
Thank you
I Will reach out if ever get into an issue.

**Comment by anandwana001** [comment]
@chrk2205 updating the wiki doc to get a better understanding. will update you today.

**Comment by anandwana001** [comment]
@chrk2205 I had added few points here, please take a look - https://github.com/oppia/oppia-android/wiki/Android-Studio-UI-based-Github-workflow#push-changes

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 6]
```diff
@@ -104,8 +107,47 @@ private val SKILL_ID_LIST = listOf(FRACTIONS_SKILL_ID_0)
 @LooperMode(LooperMode.Mode.PAUSED)
 @Config(application = QuestionPlayerActivityTest.TestApplication::class, qualifiers = "port-xxhdpi")
 class QuestionPlayerActivityTest {
-  // TODO(#503): add tests for QuestionPlayerActivity (use StateFragmentTest for a reference).
+  // TODO(#503): Add more test-cases
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt:113]: why updating this TODO?, I guess until we get to the end of this PR either we can keep this for reference or we can remove this line.

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 7]
```diff
@@ -104,8 +107,47 @@ private val SKILL_ID_LIST = listOf(FRACTIONS_SKILL_ID_0)
...
   // TODO(#1273): add tests for Hints and Solution in Question Player.
+  /**
+   * Tests for QuestionPlayerActivityTest
+   *
+   * TOOLBAR
+   * testQuestionPlayer_toolbarIsVisible
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: also check after changing configuration too. 

```
testQuestionPlayer_configChange_toolbarIsVisible
testQuestionPlayer_configCahnge_toolbarTitleIsCorrect
```

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 8]
```diff
@@ -104,8 +107,47 @@ private val SKILL_ID_LIST = listOf(FRACTIONS_SKILL_ID_0)
...
+   * testQuestionPlayer_toolbarTitleIsCorrect - match the string @string/question_player_title
+   *
+   * BACK BUTTON
+   * testQuestionPlayer_backPress_stopExplorationDialogIsDisplayed
+   * testQuestionPlayer_backPress_clickCancel_dialogIsDismissed
+   * testQuestionPlayer_backPress_clickLeave_questionPlayerIsClosed
```
Comment (anandwana001) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: also check after changing configuration too.

```
testQuestionPlayer_configChange_backPress_stopExplorationDialogIsDisplayed
testQuestionPlayer_configChange_backPress_clickCancel_dialogIsDismissed
testQuestionPlayer_configChange_backPress_clickLeave_questionPlayerIsClosed
```

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: Also, the other way round too

```
testQuestionPlayer_backPress_configChange_stopExplorationDialogIsDisplayed
testQuestionPlayer_backPress_configChange_clickCancel_dialogIsDismissed
testQuestionPlayer_backPress_configChange_clickLeave_questionPlayerIsClosed
```

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Thanks @chrk2205 
looks like we are on the right track. 

Did we cover all the cases we need to test here?

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt (hunk 6 above)
Comment (chrk2205) [reply app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt:113]: I will remove the TODO at the end of this PR.

**Comment by chrk2205** [comment]
@anandwana001 
I mentioned all the test cases I could think of.
shall I start writing the test cases??
I will include more if I find them.

about the `stateFragmentTestActivity` what purpose does it solve for `stateFragmentTest`. I mean why cannot we use the `stateFragment` directly for testing??

**Comment by anandwana001** [comment]
1. Yes, let's start writing test cases to cover all we discussed above.
2. We cannot directly test `StateFragment`, because to test we need to have a lot of pre-setup things so we had created a copy of it and testing the same on it. 

1 trick point here is, exploration in testing does come in a series like if you open Fraction exploration in testing, that series of question is fixed,
but, with the questions, they come in a random way, so I think here we do need a custom class and also a skill list to always test based on them.

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt [hunk 9]
```diff
@@ -104,8 +107,58 @@ private val SKILL_ID_LIST = listOf(FRACTIONS_SKILL_ID_0)
...
+   *
+   * []testQuestionPlayer_onConfig_backPress_stopQuestionPlayerDialogIsDisplayed
+   * []testQuestionPlayer_onConfig_backPress_clickCancel_dialogIsDismissed
+   * []testQuestionPlayer_onConfig_backPress_clickLeave_questionPlayerIsClosed
+   *
+   * [X]testQuestionPlayer_backPress_configChange_stopQuestionPlayerDialogIsDisplayed
+   * []testQuestionPlayer_backPress_configChange_clickCancel_dialogIsDismissed
+   * []testQuestionPlayer_backPress_configChange_clickLeave_questionPlayerIsClosed
+   *
+   * []testQuestionPlayer_clickToolbarNav_stopQuestionPlayerDialogIsDisplayed
```
Comment (chrk2205) [inline app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: @anandwana001 
shall I write these test cases  
`testQuestionPlayer_configChange_clickToolbarNav_stopQuestionPlayerDialogIsDisplayed`
 too?

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: yes, you can start writing all

Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt]: After writing, when you send this pr for review, please add 2 screenshots in the PR description showing all the test cases are passing on espresso and robolectric. 

How to run a test on Robolectrci - https://github.com/oppia/oppia-android/wiki/Oppia-Android-Testing#running-robolectric-tests

File: app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt (hunk 6 above)
Comment (anandwana001) [reply app/src/sharedTest/java/org/oppia/android/app/topic/questionplayer/QuestionPlayerActivityTest.kt:113]: Keeping this open as a reminder to remove the TODO, else you can remove it now.

**Review by anandwana001 — CHANGES_REQUESTED** [review]
Few things to figure out:
1. From my earlier comment - https://github.com/oppia/oppia-android/pull/2651#issuecomment-775656944 Can we test QuestionPlayer directly or not?
2. If we cannot test directly, we need to create QuestionPlayerTestActivity and all other required set up.
3. Question player load random question every time we open it, so we cannot test 4th or 5th question because we don't know what will be the first question and what will be it's answer. To solve this, we need a setup to stop randomization just for the testing purpose and load the question in a series which we know prior testing so we can write a correct test.

We need to test -> `if question player ends correctly`, then test will be like
```
Example -> 
Step 1:  launchForSkillList(SKILL_ID_LIST)
Step 2: Answer 1st question
Step 3: Answer 2nd question
Step 4: Answer 3rd question
Step 5: Question player ends correctly
```

**Comment by chrk2205** [comment]
@anandwana001 
We might need `QuestionPlayerTestActivity` as we need the list of questions before we are testing it.
but if we check in this test `testQuestionPlayer_forMisconception_showsLinkTextForConceptCard` it's choosing the wrong answer and there is a test case that chooses the correct answer too.

**Comment by anandwana001** [comment]
Could you try running this test multiple times and check if the question is the same all the time?, If yes, I think the questions are not coming in a random manner.

**Comment by chrk2205** [comment]
@anandwana001 
For the Ratio and Proportions chapter, there is only one question so it cannot be randomized (it is being used for testing) and for any other chapters, the questions are random.

**Comment by chrk2205** [comment]
> how to test if the present activity has closed?
> for `testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog_clickOk_activityCloses`
> tried using this - `intended(hasComponent(TopicActivity::class.java.name))`
> 
> and also if the dialog is disappeared `testQuestionPlayer_onBackPressed_showsStopQuestionPlayerDialog_clickCancel_dismissesDialog`
> tried this - `onView(withText(R.string.stop_exploration_dialog_title)).inRoot(isDialog()) .check(doesNotExist()) `
> thankyou

@anandwana001 @rt4914 
can I get some help with these?
Thank you

**Comment by anandwana001** [comment]
1. For closing activity, take reference from `testExplorationActivity_onBackPressed_showsStopExplorationDialog_clickLeave_closesExplorationActivity`, where we are using `explorationActivityTestRule` to check if the activity `isFinishing` or not. 

2. For checking if dialog is visible or not, `doesNotExist` is helpful and you can use it. Other way round to test is, first you can confirm dialog is visible completely and then click ok and then you can pick any view which was hidden under the dialog when dialog is on the screen, but after dialog gets dismissed, that view will be completely visible. 
For reference `testAdministratorControlsFragment_clickCancelButtonInLogoutDialog_dialogIsDismissed`.

**Comment by chrk2205** [comment]
@anandwana001 
@FareesHussain 
I just updated the branch and my current push is giving this error.
what should I do??
```
Push failed 
Bash: scripts/kotlin-lint-check.sh: No such file or directory
failed to push some refs to 'https://github.com/chrk2205/oppia-android.git'
```

**Comment by FareesHussain** [comment]
@chrk2205 From which directory are you running the command

**Comment by chrk2205** [comment]
@FareesHussain 
I just pushed the code from android studio

**Comment by FareesHussain** [comment]
> @FareesHussain
> I just pushed the code from android studio

Use `git bash` to push the code

**Comment by chrk2205** [comment]
@FareesHussain 
i am not sure what is happening 
but it is giving this error 
```
remote: Invalid username or password.
fatal: Authentication failed for https://github.com/chrk2205/oppia-android.git/
```
even though it's correct

**Comment by prayutsu** [comment]
@chrk2205 It is failing because you might have some lint errors in the committed files, if you use `git push origin <branchname>` then you will see the ktlint errors.
After you fix that you can push from the Android studio.

**Comment by chrk2205** [comment]
@prayutsu @FareesHussain 
either way it is not giving
Capture

**Comment by FareesHussain** [comment]
use the token for password

**Comment by chrk2205** [comment]
the same error showed in android studio 
Capture

**Comment by chrk2205** [comment]
@FareesHussain @prayutsu 
there is no file named as `kotlin-lint-check.sh ` in scripts
Capture

**Comment by FareesHussain** [comment]
@chrk2205 can you try to re-run the setup.sh
I guess the file name was changed recently

**Comment by chrk2205** [comment]
@anandwana001 @FareesHussain 
about the oppia github Home 
is it visible like this only for me or its an error??

image

**Comment by chrk2205** [comment]
@FareesHussain @prayutsu 
Thanks for the help it is working now.
Thank you

**Comment by chrk2205** [comment]
@FareesHussain 
about the googole_checkstyle I copied the file and pasted it in the git folder and then after push, the following error has been thrown.

`com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module TreeWalker - cannot initialize module RecordComponentName - Unable to instantiate 'RecordComponentName' class, it is also not possible to instantiate it as .RecordComponentName, RecordComponentNameCheck, .RecordComponentNameCheck. Please recheck that class name is specified as canonical name or read how to configure short name usage https://checkstyle.org/config.html#Packages. Please also rechec at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.ja Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: cannot initialize module RecordComponentName - Unable to instantiate 'RecordComponentName' class, it is also not possible to instantiate it as .RecordComponentName, RecordComponentNameCheck, .RecordComponentNameCheck. Please recheck that class name is specified as canonical name or read how to configure short name usage https://checkstyle.org/config.html#Packages. Please also recheck that provided ClassLoade at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.ja Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to instantiate 'RecordComponentName' class, it is also not possible to instantiate it as .RecordComponentName, RecordComponentNameCheck, .RecordComponentNameCheck. Please recheck that class name is specified as canonical name or read how to configure short name usage https://checkstyle.org/config.html#Packages. Please also at com.puppycrawl.tools.checkstyle.PackageObjectFactory.createModule(PackageObj Checkstyle ends with 1 errors.tyle.TreeWalker.setupChild(TreeWalker.java:116)`

**Comment by FareesHussain** [comment]
@chrk2205 Try using this file https://github.com/checkstyle/checkstyle/blob/14005e371803bd52dff429904b354dc3e72638c0/src/main/resources/google_checks.xml

**Comment by chrk2205** [comment]
@FareesHussain 
thankyou very much

**Comment by FareesHussain** [comment]
> @anandwana001 @FareesHussain
> about the oppia github Home
> is it visible like this only for me or its an error??
> 
> image

@anandwana001 PTAL

**Comment by chrk2205** [comment]
@anandwana001 
I updated the test cases.
the submit button is only visible for text-type answer questions !!
how can I access only the text type of question?

**Comment by anandwana001** [comment]
@chrk2205 @FareesHussain the wiki page is fine for me. 


[Image](https://lh3.googleusercontent.com/7k3QH2vmlNqZoL-kDDbqydtTUB1AwbXdYlFJM8oPPaC60k0en9QUm-diGhbQzEIwedhOMs2RRQSEXmvEg5sPmFX4Ug4doSx0Bgv1Mi65Bczd3m1M4X8fO9pN7okckSKB2H-Q9Wgua8BJEa_8wLt_V0PtRnh4m0LkChj1NE8EqHITXxfJMxp7XgZyYrXUZ6y2RTCz1cdNDZSoTpLbJeyznzBV7mBPND-9J2paMt03MtxXMO34c-vFr6w0NAZQxtRrNNBexr84Myk=w3227)

**Comment by anandwana001** [comment]
@chrk2205 `text type of question` Could you explain more on this?

**Comment by chrk2205** [comment]
@anandwana001 
The answer format should be in text to have a submit button.

**Comment by anandwana001** [comment]
This is something we already know while testing what format we are going to input as we know what is the series of question coming on screen. Check the StateFragmentTest or Exploration Test.
But, in our case you need to check are the question coming in random or in series.

**Comment by anandwana001** [comment]
@chrk2205 any update on this?

**Comment by chrk2205** [comment]
@anandwana001 
sorry, I had tests last week so I could not update them. and also 
I am still confused about how the test activity will help !!

**Comment by anandwana001** [comment]
@chrk2205 Can we do it without test Activity? 
Also, I suggest to check `StateFragmentTest` and `StateFragmentTestActivity`, this will help you understand why we tested using `StateFragmentTestActivity` and not `StateFragment` directly. 

But, in our case, if we can test directly `QuestionPlayerActivity`, then we are good to do.

**Comment by anandwana001** [comment]
@chrk2205 any update on this?

**Comment by anandwana001** [comment]
Closing this PR as no further update.
